### PR TITLE
V0.3.1: bug fixing, sobol attribution method.

### PR DESCRIPTION
**Pull request by fel-thomas**
Include the PRs  (#99 ,#100, #97) in master fixing various bugs in the documentation, allows to better manage edges cases for sklearn models, and introduce the Sobol attribution method.

thanks to @AntoninPoche ! ;)

### CODE REVIEW

**Review by AntoninPoche — APPROVED** [review]
LGTM !

**Review by justinplakoo — APPROVED** [review]
LGTM

File: setup.py [hunk 1]
```diff
@@ -5,15 +5,15 @@
...
 setup(
     name="Xplique",
-    version="0.3.0",
+    version="0.3.1",
     description="Explanations toolbox for Tensorflow 2",
     long_description=README,
     long_description_content_type="text/markdown",
     author="Thomas FEL",
-    author_email="thomas.fel@brown.edu",
+    author_email="thomas_fel@brown.edu",
```
Comment (thib-s) [inline setup.py:13]: is it a typo or is it deliberate ?

File: xplique/attributions/sobol/perturbations.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    ----------
+    input: tf.Tensor
+        Image to perform perturbation on.
+    sigma: int
+        Amplitude operator intensity.
+
+    Returns
+    -------
+    f: callable
+        Blur perturbation function.
```
Comment (thib-s) [inline xplique/attributions/sobol/perturbations.py:90]: typo in doc

File: xplique/attributions/sobol/sobol_attribution_method.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    @staticmethod
+    def from_string(perturbation_function: str) -> 'PerturbationFunction':
+        """
+        Restore a perturbation function from a string.
+
+        Parameters
+        ----------
+        perturbation_function
+            String indicating the perturbation function to restore, e.g 'inpainting', 'blurring'.
```
Comment (thib-s) [inline xplique/attributions/sobol/sobol_attribution_method.py:35]: could be changed to:
String indicating the perturbation function to restore: must be one of 'inpainting', 'blurring' or 'amplitude'.

**Review by thib-s — APPROVED** [review]
Just a few typos in the doc, LGTM otherwise

File: xplique/attributions/sobol/sobol_attribution_method.py (hunk 3 above)
Comment (fel-thomas) [reply xplique/attributions/sobol/sobol_attribution_method.py:35]: true, fixed ;)

File: xplique/attributions/sobol/perturbations.py (hunk 2 above)
Comment (fel-thomas) [reply xplique/attributions/sobol/perturbations.py:90]: thanks, good catch, I made this mistake for the three operator generator... ^^

**Comment by fel-thomas** [comment]
All comments have been adressed. Congrats everyone 🎉 ! ;)

**Pull request merged**